src/Components/Web/src/Forms/InputDate.cs

```diff
@@ -45,17 +69,14 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
 
         /// <inheritdoc />
         protected override string FormatValueAsString(TValue? value)
-        {
-            switch (value)
-            {
-                case DateTime dateTimeValue:
-                    return BindConverter.FormatValue(dateTimeValue, DateFormat, CultureInfo.InvariantCulture);
-                case DateTimeOffset dateTimeOffsetValue:
-                    return BindConverter.FormatValue(dateTimeOffsetValue, DateFormat, CultureInfo.InvariantCulture);
-                default:
-                    return string.Empty; // Handles null for Nullable<DateTime>, etc.
-            }
-        }
+            => value switch
+            {
+                DateTime dateTimeValue => BindConverter.FormatValue(dateTimeValue, _format, CultureInfo.InvariantCulture),
+                DateTimeOffset dateTimeOffsetValue => BindConverter.FormatValue(dateTimeOffsetValue, _format, CultureInfo.InvariantCulture),
+                DateOnly dateOnlyValue => BindConverter.FormatValue(dateOnlyValue, _format, CultureInfo.InvariantCulture),
+                TimeOnly timeOnlyValue => BindConverter.FormatValue(timeOnlyValue, _format, CultureInfo.InvariantCulture),
+                _ => string.Empty, // Handles null for Nullable<DateTime>, etc.
+            };
 
         /// <inheritdoc />
         protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out TValue result, [NotNullWhen(false)] out string? validationErrorMessage)
```

Comment:
This might not be directly related to this PR, but is there a reason we are manually checking `TValue` and calling the appropriate converter rather than relying on `BindConverter.TryConvertTo<T>()`? The only reason I can think of would be so that we can throw if `TValue` is not a date/time type. But if that's the case, couldn't we check that in the constructor and throw right away rather than waiting until `TryParseValueFromString` and performing the check every time? Relying on `TryConvertTo<T>()` would also allow us to remove ~80 LOC from this file. If there was another reason we're doing things this way, I'm completely fine with leaving it as is.

@SteveSandersonMS @pranavkm

Reply:
I'm not totally sure, but the first difference I noticed is that the logic here defines its own format string for the parsing, whereas `BindConverter.TryConvertTo<T>` doesn't let you specify any format string. Internally, `BindConverter.TryConvertTo<T>` passes `null` as the format to things like `DateTime.TryParse`.

Are you saying that the formats used by the browser happen to be parseable by `DateTime.TryParse` (and similar for `TimeOnly.TryParse`, etc.) using a `null` format in all cases, and therefore we don't need to pass a format string for parsing?

If that's the case, and everything still works if changing to `BindConverter.TryConvertTo<T>`, then that sounds great and I'd definitely support that. Doing the type compatibility check once up front also makes sense.

Reply:
It does look like the browser format is parseable by `DateTime`, etc. using a `null` format in all cases. Everything still works after changing it to use `BindConverter.TryConvertTo<T>()`. But I'm happy again to leave it as-is if we just don't want to risk breaking some weird edge case or something (although I don't foresee that happening especially if we throw in the constructor when `TValue` is something weird).

Reply:
In that case, simplifying this sounds great. The test coverage looks good so I'm not aware there will be other edge cases. If you can specifically anticipate any then it's worth discussing, but otherwise it seems fine.